core/domain/exp_jobs_one_off_test.py

```diff
@@ -739,80 +739,50 @@ def test_migration_job_skips_deleted_explorations(self):
-            init_state.update_interaction_fallbacks(self.FALLBACK)
-            exploration.add_states(['state 1'])
-            new_state = exploration.states['state 1']
-            new_state.interaction.default_outcome.dest = 'end'
-            new_state.update_interaction_fallbacks(self.FALLBACK)
-            exploration.states['end'].update_interaction_id('EndExploration')
-
-            exp_services._save_exploration(self.owner_id, exploration, '', []) # pylint: disable=protected-access
-            rights_manager.publish_exploration(self.owner_id, exploration.id)
-
-        job_id = exp_jobs_one_off.FallbackOneOffJob.create_new()
-        exp_jobs_one_off.FallbackOneOffJob.enqueue(job_id)
+        exp_services.load_demo(self.REGULAR_EXP_ID)
+
+        exp_services.load_demo(self.GADGET_EXP_ID)
+        exploration = exp_services.get_exploration_by_id(self.GADGET_EXP_ID)
+        exp_services.update_exploration(self.owner_id, self.GADGET_EXP_ID, [{
+            'cmd': exp_domain.CMD_ADD_GADGET,
+            'panel': 'bottom',
+            'gadget_dict' : {
+                'gadget_type': 'ScoreBar',
+                'gadget_name': 'NewGadget',
+                'customization_args': {
+                    'adviceObjects': {
+                        'value': [{
+                            'adviceTitle': 'b',
+                            'adviceHtml': '<p>c</p>'
+                        }]
+                    }
+                },
+                'visible_in_states': [exploration.init_state_name]
+            }
+        }], 'add a new gadget')
+
+    def test_explorations_with_gadgets_are_listed(self):
+        """Tests that explorations with gadgets are listed."""
+        job_id = exp_jobs_one_off.GadgetsOneOffJob.create_new()
+        exp_jobs_one_off.GadgetsOneOffJob.enqueue(job_id)
         self.process_and_flush_pending_tasks()
 
-        actual_values = exp_jobs_one_off.FallbackOneOffJob.get_output(job_id)
-
-        expected_values = [[
-            u'exp_id0: Introduction',
-            [u'3: <p>feedback 0</p>',
-             u'4: <p>feedback 1</p>',
-             u'5: <p>feedback 2</p>']
-        ], [
-            u'exp_id0: state 1',
-            [u'3: <p>feedback 0</p>',
-             u'4: <p>feedback 1</p>',
-             u'5: <p>feedback 2</p>']
-        ], [
-            u'exp_id1: Introduction',
-            [u'3: <p>feedback 0</p>',
-             u'4: <p>feedback 1</p>',
-             u'5: <p>feedback 2</p>']
-        ], [u'exp_id1: state 1',
-            [u'3: <p>feedback 0</p>',
-             u'4: <p>feedback 1</p>',
-             u'5: <p>feedback 2</p>']]]
-
-        actual_values.sort()
-        for index, actual_value in enumerate(actual_values):
-            self.assertEqual(actual_value.encode('ascii'),
-                             str(expected_values[index]))
+        actual_values = [
+            value.encode('ascii')
+            for value in exp_jobs_one_off.GadgetsOneOffJob.get_output(job_id)]
```

Comment:
Haha, I actually misread it too when I looked at it after seeing this -- good call. Done!